pxr/usd/usdGeom/schema.usda

```diff
@@ -803,6 +803,66 @@ class Cone "Cone" (
     )
 }
 
+class Plane "Plane" (
+    inherits = </Gprim>
+    doc = """Defines a primitive plane, centered at the origin, and is defined by
+    a cardinal axis, width, and length. It is also double-sided.
+
+    The axis of width and length are perpendicular to the plane's \\em axis:
+
+    axis  | width  | length
+    ----- | ------ | -------
+    X     | z-axis | y-axis
+    Y     | x-axis | z-axis
+    Z     | x-axis | y-axis
+
+    """
+    customData = {
+        dictionary extraPlugInfo = {
+            bool implementsComputeExtent = true
+        }
+    }
+) {
+    uniform bool doubleSided = true (
+        doc = """Planes are double-sided by default, clients may also supported
+        single-sided planes.
+
+        \\sa UsdGeomGprim::GetDoubleSidedAttr()"""
+        )
+    double width = 2.0 (
+        doc = """The width of the plane, which aligns to the x-axis when \\em axis is
+        'Z' or 'Y', or to the z-axis when \\em axis is 'X'.  If you author \\em width 
+        you must also author \\em extent.
+
+        \\sa UsdGeomGprim::GetExtentAttr()"""
+    )
+    double length = 2.0 (
+        doc = """The length of the plane, which aligns to the y-axis when \\em axis is
+        'Z' or 'X', or to the z-axis when \\em axis is 'Y'.  If you author \\em length 
+        you must also author \\em extent.
+
+        \\sa UsdGeomGprim::GetExtentAttr()"""
+    )
+    uniform token axis = "Z" (
+        allowedTokens = ["X", "Y", "Z"]
+        doc = """The axis along which the surface of the plane is aligned. When set
+        to 'Z' the plane is in the xy-plane; when \\em axis is 'X' the plane is in 
+        the yz-plane, and when \\em axis is 'Y' the plane is in the xz-plane.
+
+        \\sa UsdGeomGprim::GetAxisAttr()."""
+    )
+    float3[] extent = [(-1.0, -1.0, 0.0), (1.0, 1.0, 0.0)] (
+        doc = """Extent is re-defined on Plane only to provide a fallback
+        value. \\sa UsdGeomGprim::GetExtentAttr()."""
+    )
+    texCoord2d[] primvars:st = [(1.0,1.0),(0.0,1.0),(0.0,0.0),(1.0,0.0)] (
+        doc = """The default st coordinates for texture maps.
+
+        \\sa UsdGeomGprim::GetPrimvarsStAttr()"""
+        interpolation = "varying"
+    )
+}
```

Comment:
We think the "primvars:st" attribute should be removed from this schema. We're uncomfortable hard-coding this name both because everybody has a different "uv" naming scheme; declaring "primvars:" values on base schema types has caused headaches in the past; and this is assuming a certain tessellation.

Our suggestion instead is that, somewhere in the prim comments, we mention that the plane is parameterized as s -> (0, 1) along width axis, t -> (0, 1) along length axis, and usdImaging will generate a primvars:st texcoord with this parameterization. And I guess since it's centered, that would map (0,1) -> (-width * 0.5, width * 0.5) on the appropriate axis.

If folks are ok with the primvar being hardcoded to "st", we're good; if they need to be able to name it, we could add a "token textureCoordinate" attribute in the future.

We could merge this PR without the primvars:st attribute and any of the suggestions above first, then look into implementing the additional behavior afterwards. If you (or anyone) would be interested in implementing the proposed changes in another PR after this landed, we'd gladly take a look. Otherwise it's something we can add to our queue though realistically we might not be able to get to that for a while.

Reply:
That makes sense. Will watch the discussion on usd-interest, but more than happy to remove the texture coordinates for now.